modules/database/src/adapters/sequelize-adapter/utils/schema.ts

```diff
@@ -115,3 +118,140 @@ export const sqlTypesProcess = (
   }
   return idField ?? '_id';
 };
+
+export async function getTransactionAndParsedQuery(
+  transaction: Transaction | undefined,
+  query: string | ParsedQuery,
+  sequelize: Sequelize,
+): Promise<{ t: Transaction; parsedQuery: ParsedQuery; transactionProvided: boolean }> {
+  let t: Transaction | undefined = transaction;
+  const transactionProvided = transaction !== undefined;
+  let parsedQuery: ParsedQuery;
+  if (typeof query === 'string') {
+    parsedQuery = JSON.parse(query);
+  } else {
+    parsedQuery = query;
+  }
+  if (parsedQuery.hasOwnProperty('$set')) {
+    parsedQuery = parsedQuery['$set'];
+  }
+  if (isNil(t)) {
+    t = await sequelize.transaction({ type: Transaction.TYPES.IMMEDIATE });
+  }
+  return { t, parsedQuery, transactionProvided };
+}
+
+export function processPushOperations(
+  parentDoc: any,
+  push: any,
+  extractedRelations: any,
+) {
+  for (const key in push) {
+    if (extractedRelations[key]) {
+      if (!Array.isArray(extractedRelations[key])) {
+        throw new Error(`Cannot push in non-array field: ${key}`);
+      }
+      let modelName = key.charAt(0).toUpperCase() + key.slice(1);
+      if (push[key]['$each']) {
+        if (!modelName.endsWith('s')) {
+          modelName = modelName + 's';
+        }
+        parentDoc[`add${modelName}`](push[key]['$each'], parentDoc._id);
+      } else {
+        const actualRel = key.charAt(0).toUpperCase() + key.slice(1);
+        parentDoc[`add${actualRel}Id`](push[key], parentDoc._id);
+      }
+      continue;
+    }
+    if (push[key]['$each']) {
+      parentDoc[key] = [...parentDoc[key], ...push[key]['$each']];
+    } else {
+      parentDoc[key] = [...parentDoc[key], push[key]];
+    }
+  }
+}
+
+export function compileSchema(
+  schema: ConduitDatabaseSchema,
+  registeredSchemas: Map<string, ConduitSchema>,
+  sequelizeModels: { [key: string]: any },
+): ConduitDatabaseSchema {
+  let compiledSchema = JSON.parse(JSON.stringify(schema));
+  validateFieldConstraints(compiledSchema);
+  (compiledSchema as any).fields = JSON.parse(JSON.stringify(schema.compiledFields));
+  if (registeredSchemas.has(compiledSchema.name)) {
+    if (compiledSchema.name !== 'Config') {
+      compiledSchema = validateFieldChanges(
+        registeredSchemas.get(compiledSchema.name)!,
+        compiledSchema,
+      );
+    }
+    delete sequelizeModels[compiledSchema.collectionName];
+  }
+  return compiledSchema;
+}
+type Relation = {
+  type: 'Relation';
+  model: string;
+  required?: boolean;
+  select?: boolean;
+};
+
+type ExtractedRelations = {
+  [p: string]: Relation | Relation[];
+};
+export async function registerAndResolveRelatedSchemas(
+  schema: ConduitDatabaseSchema,
+  registeredSchemas: Map<string, ConduitSchema>,
+  extractedRelations: ExtractedRelations,
+  models: { [name: string]: any },
+) {
+  registeredSchemas.set(schema.name, Object.freeze(JSON.parse(JSON.stringify(schema))));
```

Comment:
registeredSchemas.set should remain in its original position since it's  not part of the relation processing.